common/djangoapps/philu_commands/management/commands/send_ondemand_reminder_emails.py

```diff
@@ -0,0 +1,191 @@
+                    log.info(error)
+                    continue
+
+                # If user hasn't enable email preferences for on demand course, no need to go further.
+                if not get_email_pref_on_demand_course(user, course.id):
+                    continue
+
+                course_chapters = modulestore().get_items(
+                    course.id,
+                    qualifiers={'category': 'course'}
+                )
+
+                course_deadline = get_suggested_course_deadline(enrollment.created.date(), course_chapters[0].children)
+
+                # Get all user submission in descending order by date
+                response_submissions = Submission.objects.filter(
+                    student_item__student_id=anonymous_user.anonymous_user_id,
+                    student_item__course_id=course.id.to_deprecated_string()).order_by('-created_at')
+
+                # Check if user has submitted last modules graded oras or not. If yes not need to send email OR
+                # If user's submission gets equal to graded oras count than don't need to continue..
+                if (last_module_oras and check_for_last_module_submission(last_module_oras, anonymous_user)) or \
+                        len(response_submissions) == graded_oras_count:
+                    continue
+
+                latest_submission = response_submissions.first()
+                # Count to keep track of how many times user shows an inactivity for "INACTIVITY_REMINDER_DAYS"
+                time_passed_count = 0
+
+                if len(response_submissions) == 0 and has_time_passed_since_activity(enrollment.created.date(), today):
+                    send_reminder_email(user, course, course_deadline)
+                    continue
+                elif len(response_submissions) > 0:
+                    # Check for latest submission entry from submission table if the difference of created date and
+                    # today is equals to "INACTIVITY_REMINDER_DAYS" days this means that email should be send to user
+                    # but before that we need to check is it the first time user shows an inactivity for
+                    # "INACTIVITY_REMINDER_DAYS" days or not if yes than send email else means that emails has already
+                    #  been sent to user so no need to send it again.
+                    if has_time_passed_since_activity(latest_submission.created_at.date(), today):
+                        last_response_time = latest_submission.created_at.date()
+                        # We have checked first entry separately so starting from second index.
+                        for index_response, response in enumerate(response_submissions[1:]):
+                            if has_time_passed_since_activity(response.created_at.date(), last_response_time):
+                                time_passed_count += 1
+                        # If user haven't been inactive in past before today than send email.
```

Comment:
we might be missing a line here:
`last_response_time = response.created_at.date()`